mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -551,6 +563,21 @@ object ALSModel extends MLReadable[ALSModel] {
       model
     }
   }
+
+  /** select top indices based on values. */
+  private[recommendation] class TopSelector(val values: Array[Float]) {
```

Comment:
Does this need to be a class? looks like this code is called once. May be less code/indirection

Reply:
I will consider whether a class is needed.